rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbCopyAttributesToEntityViewNodeTest.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Test
+    public void givenAttributesValidityPeriodOutOfStartDateAndEndDate_whenOnMsg_thenDoNothing() {
+        EntityViewId entityViewId = new EntityViewId(UUID.fromString("d117f1a4-24ea-4fdd-b94e-5a472e99d925"));
+        EntityView entityView = new EntityView(entityViewId);
+        entityView.setStartTimeMs(Instant.now().minus(2, ChronoUnit.DAYS).toEpochMilli());
+        entityView.setEndTimeMs(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+
+        mockEntityViewService(entityView);
+
+        TbMsg msg = TbMsg.newMsg(
+                ATTRIBUTES_DELETED, DEVICE_ID, new TbMsgMetaData(Map.of("scope", AttributeScope.CLIENT_SCOPE.name())),
+                "{\"attributes\": [\"attribute1\"]}");
+
+        node.onMsg(ctxMock, msg);
+
+        verify(entityViewServiceMock).findEntityViewsByTenantIdAndEntityIdAsync(eq(TENANT_ID), eq(DEVICE_ID));
+        verify(ctxMock).ack(eq(msg));
+    }
+
+    @Test
+    public void givenEmptyMetadata_whenOnMsg_thenThrowsException() {
+        TbMsg msg = TbMsg.newMsg(
+                ATTRIBUTES_UPDATED, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+
+        node.onMsg(ctxMock, msg);
+
+        ArgumentCaptor<Throwable> throwableCaptor = ArgumentCaptor.forClass(Throwable.class);
+        verify(ctxMock).tellFailure(eq(msg), throwableCaptor.capture());
+        assertThat(throwableCaptor.getValue()).isInstanceOf(IllegalArgumentException.class).hasMessage("Message metadata is empty");
+    }
+
+    @ParameterizedTest
+    @EnumSource(TbMsgType.class)
+    public void givenUnsupportedMsgType_whenOnMsg_thenTellFailure(TbMsgType msgType) {
+        TbMsg msg = TbMsg.newMsg(
+                msgType, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+
+        if (msg.isTypeOneOf(ATTRIBUTES_UPDATED, ATTRIBUTES_DELETED,
+                ACTIVITY_EVENT, INACTIVITY_EVENT, POST_ATTRIBUTES_REQUEST)) {
+            return;
+        }
+
+        node.onMsg(ctxMock, msg);
+
+        ArgumentCaptor<Throwable> throwableCaptor = ArgumentCaptor.forClass(Throwable.class);
+        verify(ctxMock).tellFailure(eq(msg), throwableCaptor.capture());
+        assertThat(throwableCaptor.getValue()).isInstanceOf(IllegalArgumentException.class).hasMessage("Unsupported msg type [" + msgType + "]");
+    }
+
+    private EntityView getEntityView() {
+        EntityViewId entityViewId = new EntityViewId(UUID.fromString("a2109747-d1f4-475a-baaa-55f5d4897ad8"));
+        EntityView entityView = new EntityView(entityViewId);
+        entityView.setStartTimeMs(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+        entityView.setEndTimeMs(Instant.now().plus(1, ChronoUnit.DAYS).toEpochMilli());
+        AttributesEntityView attributes = new AttributesEntityView(List.of("attribute1"), Collections.emptyList(), Collections.emptyList());
+        entityView.setKeys(new TelemetryEntityView(Collections.emptyList(), attributes));
+        return entityView;
+    }
```

Comment:
This can be moved to constants:


```
private final long ENTITY_VIEW_START_TS = Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli();
private final long ENTITY_VIEW_END_TS = Instant.now().plus(1, ChronoUnit.DAYS).toEpochMilli();
private final AttributesEntityView CLIENT_ATTRIBUTES = new AttributesEntityView(List.of("attribute1"), Collections.emptyList(), Collections.emptyList());
private final TelemetryEntityView TELEMETRY_ENTITY_VIEW = new TelemetryEntityView(Collections.emptyList(), CLIENT_ATTRIBUTES);
```

and this:

```
private final EntityViewId ENTITY_VIEW_ID = new EntityViewId(UUID.fromString("a2109747-d1f4-475a-baaa-55f5d4897ad8"));
```

Finally method will be looks like this:

```
  private EntityView getEntityView() {
      EntityView entityView = new EntityView(ENTITY_VIEW_ID);
      entityView.setStartTimeMs(ENTITY_VIEW_START_TS);
      entityView.setEndTimeMs(ENTITY_VIEW_END_TS);
      entityView.setKeys(TELEMETRY_ENTITY_VIEW);
      return entityView;
  }
```